src/cargo/ops/cargo_package.rs

```diff
@@ -258,67 +258,91 @@ fn build_ar_list(
             contents: FileContents::Generated(GeneratedFile::VcsInfo(vcs_info)),
         });
     }
+
     if let Some(license_file) = &pkg.manifest().metadata().license_file {
-        let license_path = Path::new(license_file);
-        let abs_license_path = paths::normalize_path(&pkg.root().join(license_path));
-        if abs_license_path.exists() {
-            match abs_license_path.strip_prefix(&pkg.root()) {
-                Ok(rel_license_path) => {
-                    if !result.iter().any(|ar| ar.rel_path == rel_license_path) {
-                        result.push(ArchiveFile {
-                            rel_path: rel_license_path.to_path_buf(),
-                            rel_str: rel_license_path
-                                .to_str()
-                                .expect("everything was utf8")
-                                .to_string(),
-                            contents: FileContents::OnDisk(abs_license_path),
-                        });
-                    }
+        check_for_file_and_add("license-file", license_file, pkg, &mut result, ws)?
+    }
+
+    if let Some(readme_file) = &pkg.manifest().metadata().readme {
+        check_for_file_and_add("readme", readme_file, pkg, &mut result, ws)?
+    }
+
+    result.sort_unstable_by(|a, b| a.rel_path.cmp(&b.rel_path));
+
+    Ok(result)
+}
+
+fn check_for_file_and_add(
```

Comment:
Ideally this refactor would be done in a separate commit than the adding of readme support

Reply:
I can move it to a separate commit if we keep it

Reply:
Yes, I'm liking the refactor.  We can make it the first commit of this PR.